### PR TITLE
RT major cleanup

### PR DESCRIPTION
**Pull request by bhavesh-desai-scratch**
- Strongly Type epics and actions and the store.
- Remove unnecessary bloat in the code especially around pricing and store creation
- Use redux-epic DI instead of High order functions for epics
- Increase print width as Type definitions use more space
- remove extensive references to openfin all over the code base. Openfin functionality should isolated and abstracted away.

### CODE REVIEW

File: src/client/src/ActionHelper.ts [hunk 1]
```diff
@@ -0,0 +1,53 @@
+/**
+ * Typesctipt definitions for creating actions and action creators.
```
Comment (fdore) [inline src/client/src/ActionHelper.ts]: typo => Typescript

File: src/client/src/openfinEpics.ts [hunk 2]
```diff
@@ -0,0 +1,70 @@
...
+  return {
+    CurrencyPair: price.symbol,
+    SpotRate: spotRate,
+    Direction: direction,
+    Notional: notional,
+    DealtCurrency: price.currencyPair.base
+  }
+}
+
+export const closePositionEpic: ApplicationEpic = (action$, state$, { openFin }) =>
```
Comment (fdore) [inline src/client/src/openfinEpics.ts]: state$ => store

File: src/client/src/system/AutoBahnTypeExtensions.ts [hunk 3]
```diff
@@ -1,4 +1,4 @@
-import { Connection, Session } from 'autobahn'
+//import { Connection, Session } from 'autobahn'
```
Comment (fdore) [inline src/client/src/system/AutoBahnTypeExtensions.ts]: can remove

File: src/client/src/openfinEpics.ts (hunk 2 above)
Comment (bhavesh-desai-scratch) [reply src/client/src/openfinEpics.ts]: Nope. It has changed to $state. state.getStore will be deprecated in the next version!

At the moment $state is both.

File: src/client/src/ui/analytics/AnalyticsContainer.tsx [hunk 4]
```diff
@@ -1,13 +1,9 @@
-import * as PropTypes from 'prop-types'
 import * as React from 'react'
```
Comment (frankwallis) [inline src/client/src/ui/analytics/AnalyticsContainer.tsx]: Consider turning on `esModuleInterop` and `allowSyntheticDefaultImports` in tsconfig.json, then you can import React and (and other commonjs modules) like so:
```
import React from 'react'
```

File: src/client/src/ui/spotTile/notional/NotionalContainer.tsx [hunk 5]
```diff
@@ -18,14 +18,11 @@ interface NotionalContainerDispatchProps {
   onNotionalInputChange: (payload: NotionalUpdate) => void
 }
 
-type NotionalContainerProps = NotionalContainerOwnProps &
-  NotionalContainerStateProps &
-  NotionalContainerDispatchProps
+type NotionalContainerProps = NotionalContainerOwnProps & NotionalContainerStateProps & NotionalContainerDispatchProps
 
 class NotionalContainer extends React.Component<NotionalContainerProps, any> {
```
Comment (frankwallis) [inline src/client/src/ui/spotTile/notional/NotionalContainer.tsx]: `any` parameter is no longer needed

File: src/client/src/ui/spotTile/spotTileControls/SpotTileControls.tsx [hunk 6]
```diff
@@ -3,47 +3,32 @@ import * as React from 'react'
...
   displayCurrencyChart: () => void
   onPopoutClick: () => void
   undockTile: () => void
 }
 
-export default class SpotTileControls extends React.Component<
-  SpotTileControlsProps,
-  {}
-> {
+export default class SpotTileControls extends React.Component<SpotTileControlsProps, {}> {
```
Comment (frankwallis) [inline src/client/src/ui/spotTile/spotTileControls/SpotTileControls.tsx]: `{}` parameter is no longer needed

File: src/client/src/ui/spotTile/SpotTileContainer.tsx [hunk 7]
```diff
@@ -57,39 +40,30 @@ interface SpotTileContainerStateProps {
...
+  displayCurrencyChart: (symbol: string) => () => void
+  onNotificationDismissedClick: (symbol: string) => () => void
 }
 
-type SpotTileContainerProps = SpotTileContainerOwnProps &
-  SpotTileContainerStateProps &
-  SpotTileContainerDispatchProps
+type SpotTileContainerProps = SpotTileContainerOwnProps & SpotTileContainerStateProps & SpotTileContainerDispatchProps
 
 class SpotTileContainer extends React.Component<SpotTileContainerProps, any> {
```
Comment (frankwallis) [inline src/client/src/ui/spotTile/SpotTileContainer.tsx]: `any` parameter is no longer needed

File: src/client/src/ui/analytics/AnalyticsContainer.tsx (hunk 4 above)
Comment (bhavesh-desai-scratch) [reply src/client/src/ui/analytics/AnalyticsContainer.tsx]: Wow. Thanks!

**Pull request merged**